Evaluation benchmark (#106)
chhoumann/p7: report/sections/Chapter 02 - Preliminaries/preliminaries.tex

> BenchmarkDotnet\cite{Benchmarkdotnet} is a benchmarking framework allowing time measurements of operations performed in dotnet runtimes. 

Reviewer: ```suggestion
BenchmarkDotNet\cite{Benchmarkdotnet} is a benchmarking framework for measuring the speed of operations performed in .NET runtimes. 
```

Reviewer 2: Kinda implies that it benchmarks operations in the .NET runtime, not using .NET to benchmark operations?

It allows for the parameterization of measured operations and provides ways to introduce warm-up operation and adjustments. 

> We use this framework to create a stress test environment that simulates stressful client behavior, and measures the response times for the system under test.

Reviewer: ```suggestion
We use this framework to create a stress test environment simulating stressful client behavior.
This way, we can measure the response times for the system under test.
```

> We want to use results from the stress tests and benchmarks to determine whether the built-in capabilities of the HTTP server library Rocket suits the use case of the system or if the platform benefits from a multithreaded approach.

Reviewer: ```suggestion
We want to use results from the stress tests and benchmarks to determine whether the built-in capabilities of the HTTP server library Rocket suits the use case of the system or if the platform benefits from a multithreaded approach.
```
This needs to be rewritten. The bottleneck was never Rocket. The bottleneck was handling one request at a time. The only reason we switched to axum is because it was easier to share states among endpoints - not because it's faster.
